spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/zuul/SampleZuulProxyApplicationTests.java

```diff
@@ -146,7 +146,7 @@ public void simpleHostRouteWithEncodedQuery() {
 				"http://localhost:" + this.port + "/self/query?foo={foo}", HttpMethod.GET,
 				new HttpEntity<>((Void) null), String.class, "weird#chars");
 		assertEquals(HttpStatus.OK, result.getStatusCode());
-		assertEquals("/query?foo=weird#chars", result.getBody());
+		assertEquals("/query?foo=weird%23chars", result.getBody());
```

Comment:
I'm worried that this test changed. The `#` char has meaning in a url, if it's changed, I'm not sure I can accept this.